### PR TITLE
Add Producer and PushConsumer

### PR DESCRIPTION
**Pull request by ShannonDing**
Support Features:
1. Send Messages Sync
2. Push Consumer

See more in RIP4 : https://docs.google.com/document/d/1kngd9fyzgG6g48mMNGgnvm1Cs0f-emik2_ROv_a1ZvY

### CODE REVIEW

File: changelog [hunk 1]
```diff
@@ -0,0 +1,3 @@
+version 1.0.0 @2018.10.29
```
Comment (wenfengwang) [inline changelog]: version of 1.0.0 means to stable, but the repo being developed, is it better to use 0.0.1 or 0.1.0? furthermore, i suggest we use [Semantic Versioning](https://semver.org/) to control our version-release, what's about you?

File: src/client/pushconsumer.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+func CreatePushConsumer(groupId string) (consumer PushConsumer) {
+	consumer = C.CreatePushConsumer(C.CString(groupId))
+	return consumer;
+}
+func DestroyPushConsumer(consumer PushConsumer) {
+	consumer = C.DestroyPushConsumer(consumer.(*C.struct_CPushConsumer))
+	return
+}
+func StartPushConsumer(consumer PushConsumer) int {
```
Comment (wenfengwang) [inline src/client/pushconsumer.go]: Is object-oriented Coding more better ? such as:
```go
func (p *Producer) Start() {...}

func (p *Producer) Shutdown() {...}
```
and so.

File: src/client/producer.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+//#include "CProducer.h"
+import "C"
+
+//type Producer C.struct_CProducer
+//type Message C.struct_CMessage
+//type SendResult C.struct_CSendResult
+//type PushConsumer C.struct_CPushConsumer
+//type MessageExt C.struct_CMessageExt
+
+func CreateMessage(topic string)(msg Message){
```
Comment (wenfengwang) [inline src/client/producer.go]: suggestion that use native message definition in go to avoid SetXXX method.

**Comment by wenfengwang** [comment]
I can improve the code base after the PR will have merged :)~

File: src/client/pushconsumer.go (hunk 2 above)
Comment (ShannonDing) [reply src/client/pushconsumer.go]: The basic principle is that the simpler the better, and it is better to keep the style  being consistent with the C apis, I think.

File: changelog (hunk 1 above)
Comment (ShannonDing) [reply changelog]: the version of Go SDK is consistent with C SDK now.

File: src/client/pushconsumer.go (hunk 2 above)
Comment (zjykzk) [reply src/client/pushconsumer.go]: why not be consistent with the golang style

programing with the c style using golang that maybe not good idea

Comment (wenfengwang) [reply src/client/pushconsumer.go]: there may be no one write golang code with C style. we should definitely keep C API clean and simple, but we should provide friendly API for users in go, too. Actually, we can wrap C API in go in order to expose Golang-Style API,  i had cloned code from @ShannonDing codebase and refactored it as soon as possible. the codebase will be consistent with the golang style after refactoring.

File: src/sample/sample_producer.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package main
+
+import "fmt"
+import "../client"
```
Comment (zjykzk) [inline src/sample/sample_producer.go]: pull path import is better

File: src/test/client_test.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+package client_test
+
+import (
+    "fmt"
+    "testing"
+    "../client"
+)
+
+func TestVersion(test *testing.T){
+    fmt.Println("-----TestGetVersion Start----")
```
Comment (zjykzk) [inline src/test/client_test.go]: use the `test.Log` in the ut, instead of `fmt.Println`

Comment (ShannonDing) [reply src/test/client_test.go]: Good Idea.

**Pull request merged**